internal/resources/fabric/service_token/resource.go

```diff
@@ -127,16 +143,27 @@ func waitForStability(uuid string, meta interface{}, d *schema.ResourceData, ctx
 			if err != nil {
 				return "", "", equinix_errors.FormatFabricError(err)
 			}
-			return serviceToken, string(serviceToken.GetState()), nil
+			currentState := string(serviceToken.GetState())
+			return serviceToken, currentState, nil
 		},
-		Timeout:    tieout,
+		Timeout:    timeout,
 		Delay:      30 * time.Second,
 		MinTimeout: 30 * time.Second,
 	}
 
-	_, err := stateConf.WaitForStateContext(ctx)
+	inter, err := stateConf.WaitForStateContext(ctx)
+	var serviceToken *fabricv4.ServiceToken
 
-	return err
+	if err != nil {
+		log.Printf("[ERROR] Error while waiting for state: %v", err)
```

Comment:
Should improve the error message. It doesn't say what state the method is waiting for.